types/wordpress__blocks/index.d.ts

```diff
@@ -503,3 +504,20 @@ export type Transform<T extends Record<string, any> = Record<string, any>> =
     | TransformPrefix<T>
     | TransformRaw<T>
     | TransformShortcode<T>;
+
+export type BlockVariationScope = 'block' | 'inserter' | 'transform';
+
+export interface BlockVariation<TAttributes extends Record<string, any> = any> {
+    name: string;
+    title: string;
+    description?: string;
+    category?: string;
+    icon?: BlockIcon;
+    isDefault?: boolean;
+    attributes?: Partial<TAttributes>;
+    innerBlocks?: any[];
+    example?: Record<string, any>;
+    scope?: BlockVariationScope[];
```

Comment:
what's the purpose of making `TAttributes` if we only reference `Partial<TAttributes>`? I suspect we're wanting to tie it in as a subset of the available block attributes for a given block, but if that's the case we're not actually tying it in here. I can see a reason to do it, but as they are written I'm curious if it's confusing more than helpful?

Reply:
You are right, actually if something was required for the base block we'd require it here as well so it might have even been just wrong 🤦  - will change it.